src/Microsoft.AspNet.Mvc.TagHelpers/CacheTagHelper.cs

```diff
@@ -346,11 +351,22 @@ private void AddVaryByRouteKey(StringBuilder builder)
             }
         }
 
-        private static IEnumerable<string> Tokenize(string value)
+        private static IList<string> Tokenize(string value)
         {
-            return value.Split(AttributeSeparator, StringSplitOptions.RemoveEmptyEntries)
-                        .Select(token => token.Trim())
-                        .Where(token => token.Length > 0);
+            var values = value.Split(AttributeSeparator, StringSplitOptions.RemoveEmptyEntries);
```

Comment:
Does this happen every request? We really don't want to do `string.Split`

Can you log an issue to follow up on this guy? splitting and trimming a comma-separated list seems like it should be first class and handled by codegen.